### PR TITLE
Feat: Support "Grouped" Menu Items

### PR DESCRIPTION
**Pull request by eljefe223**
# Pull Request

## 📖 Description

This PR adds support for grouping menu-items.

### General

- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: change/@microsoft-fast-foundation-cc5d0b8b-12a2-4fea-a8eb-dff8699bc3dd.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "Feat: support grouped menu-items",
+  "packageName": "@microsoft/fast-foundation",
+  "email": "jes@microsoft.com",
+  "dependentChangeType": "patch"
```
Comment (scomea) [inline change/@microsoft-fast-foundation-cc5d0b8b-12a2-4fea-a8eb-dff8699bc3dd.json:6]: prerelease?

File: packages/web-components/fast-foundation/src/menu/menu.ts [hunk 2]
```diff
@@ -291,6 +299,13 @@ export class FASTMenu extends FASTElement {
         );
     };
 
+    /**
+     * check if the item is a grouped menu item
+     */
+    protected isGroupedMenuItemElement = (el: Element): el is HTMLElement => {
+        return (el.getAttribute("role") as string) === "group";
```
Comment (scomea) [inline packages/web-components/fast-foundation/src/menu/menu.ts:306]: Do you need to cast to string?

**Review by scomea — APPROVED** [review]
Ran it, seems good.

**Comment by janechu** [comment]
Closing this, due to #6951 we are only putting in fixes or critical features.

**Pull request closed without merging**